app/questionnaire/placeholder_transforms.py

```diff
@@ -292,3 +298,33 @@ def _create_hyperlink(href: str, link_text: str) -> str:
 
     def list_item_count(self, list_to_count: Optional[Sized]) -> int:
         return self._operations.evaluate_count(list_to_count)
+
+    def option_label_from_value(self, value: str, answer_id: str) -> str:
+
+        answers = self.schema.get_answers_by_answer_id(answer_id)
+        option_label: str = next(
+            (
+                options["label"]
+                for answer in answers
+                for options in answer["options"]
+                if value in options["value"]
+            ),
+            "",
+        )
+
+        if isinstance(option_label, str):
+            label = option_label
+
+        elif isinstance(option_label, dict):
+            parser = placeholder_parser.PlaceholderParser(
+                language=self.locale,
+                answer_store=AnswerStore(),
+                list_store=ListStore(),
+                metadata=ImmutableDict({}),
+                response_metadata={},
+                schema=self.schema,
+            )
+            placeholder_text = option_label["text"].replace("{", "").replace("}", "")
```

Comment:
We shouldn't need to do anything like this. I think what you need is `PlaceholderRenderer` which is responsible for getting the rendered text. It should to be initialised at the class level to prevent initialising it multiple times within the same block. It might be cleaner to pass the class an initialised placeholder renderer than pass it all the stores and then initialise itself.

`RuleEvaluator` can create an instance of `PlaceholderRender` and pass it to `Operations`, you will still need to do the same here and have access to `PlaceholderRender` from `PlaceholderTransforms`